src/test/scala/chiseltest/tests/BasicTest.scala

```diff
@@ -30,6 +31,40 @@ class BasicTest extends FlatSpec with ChiselScalatestTester with Matchers {
     }
   }
 
+  it should "test record partial poke" in {
+    val typ = new CustomBundle("foo" -> UInt(32.W), "bar" -> UInt(32.W))
+    test(new PassthroughModule(typ)) { c =>
+      c.in.pokePartial(typ.Lit(
+        _.elements("foo") -> 4.U
+      ))
+      c.out.expectPartial(typ.Lit(
+        _.elements("foo") -> 4.U
+      ))
+      c.in.pokePartial(typ.Lit(
+        _.elements("bar") -> 5.U
+      ))
+      c.out.expect(typ.Lit(
+        _.elements("foo") -> 4.U,
+        _.elements("bar") -> 5.U
+      ))
+    }
+  }
+
+  it should "fail on partial expect mismatch" in {
+    val typ = new CustomBundle("foo" -> UInt(32.W), "bar" -> UInt(32.W))
+    assertThrows[exceptions.TestFailedException] {
+      test(new PassthroughModule(typ)) { c =>
+        c.in.pokePartial(typ.Lit(
+          _.elements("foo") -> 4.U
+        ))
+        c.out.expect(typ.Lit(
+          _.elements("foo") -> 4.U,
+          _.elements("bar") -> 5.U
```

Comment:
I think this actually depends on undefined initialization behavior?

It might also be worth adding a test for partial expect failure (as opposed to full expect failure)?